### ISSUE TITLE
No inference from custom csv model

### ISSUE THREAD

**Issue by GaurangGavai** [xuannianz/EfficientDet#117]
Hey! Thanks for your awesome work!

I've been trying to train a custom b0 model on my own data. I used the following two commands for step 1 and 2:

python3 train.py --snapshot imagenet --phi 0 --gpu 0 --random-transform --compute-val-loss --freeze-backbone --batch-size 32 --steps 1000 csv ../../toolbox+hands_trunc_effdet/toolkitbox+handtrunc.csv ../../toolbox+hands_trunc_effdet/toolkitbox+handtrunc_names.csv --val-annotations-path ../../toolbox+hands_trunc_effdet/toolkitbox+handtrunc_val.csv 

python3 train.py --snapshot checkpoints/2020-03-23/csv_50_0.7909_0.6005.h5 --phi 0 --gpu 0 --random-transform --compute-val-loss --freeze-bn --batch-size 4 --steps 10000 csv ../../toolbox+hands_trunc_effdet/toolkitbox+handtrunc.csv ../../toolbox+hands_trunc_effdet/toolkitbox+handtrunc_names.csv --val-annotations-path ../../toolbox+hands_trunc_effdet/toolkitbox+handtrunc_val.csv

The final output from step 2 is as follows:
Epoch 50/50
 9999/10000 [============================>.] - ETA: 0s - loss: 0.1808 - regression_loss: 0.1704 - classification_loss: 0.0104Epoch 1/50
 1616/10000 [===>..........................] - ETA: 6:01 - loss: 0.1154 - regression_loss: 0.1125 - classification_loss: 0.0028num_fp=30787.0, num_tp=8629.0
3249 instances of class toolkitbox with average precision: 1.0000
6359 instances of class hand with average precision: 0.8145
mAP: 0.9072

10000/10000 [==============================] - 1981s 198ms/step - loss: 0.1808 - regression_loss: 0.1704 - classification_loss: 0.0104 - val_loss: 0.1154 - val_regression_loss: 0.1126 - val_classification_loss: 0.0028

From the mAP, seems like the classes have been learned well. However there is no inference when I try inference.py on any test images (even images from the training set). I've set weighted_bifpn as both True and False but neither has helped. Please advise?

**Comment by xuannianz** [comment]
Hi @GaurangGavai ,did you train using the latest code? Recently the model arch has changed to sync with the official implementation. By the way, weighted_bifpn should always be same during training and testing.

**Comment by GaurangGavai** [comment]
I trained using the model from March 5, has it been changed since then?

**Comment by xuannianz** [comment]
Yes, the model arch was changed on Mar 21. So if your model was trained before that you should use old code.

**Comment by GaurangGavai** [comment]
I'm sorry I meant I am using the old code from March 5 and I did train the model using that same code as well but it is not working. I'm trying to train now using the new code, but is there any other reason you think the old code + old model may not be working?

**Comment by xuannianz** [comment]
Sorry, I have no idea. Just make sure your parameters of inference are corresponding with those in training.

**Comment by GaurangGavai** [comment]
Just wanted to let you know, using the new repo from March 21 fixed the issue, thanks!

**Comment by fiyerowei** [comment]
I face the same problem after using the latest code.
The training process seems no problem.
however there's no inference result. (nomatter how I try )
could any one give some suggestion about the no inference at all phenamena?
